ext/search_kit/Civi/Api4/Action/SearchDisplay/EmailReport.php

```diff
@@ -0,0 +1,161 @@
+      }
+
+      $contactIDs = explode(',', $this->contactID);
+
+      $messageTemplates = new \CRM_Core_DAO_MessageTemplate();
+      $messageTemplates->id = $this->templateID;
+
+      if (!$messageTemplates->find(TRUE)) {
+        throw new \API_Exception('Could not find template with ID: ' . $this->templateID);
+      }
+
+      [$defaultFromName, $defaultFromEmail] = \CRM_Core_BAO_Domain::getNameAndEmail();
+      $from = "\"$defaultFromName\" <$defaultFromEmail>";
+
+      $returnValues = [];
+      for ($i = 0; $i < count($contactIDs); $i++) {
+        $contactId = $contactIDs[$i];
+        $contact = \Civi\Api4\Contact::get(TRUE)
+          ->addSelect('*', 'email_primary.email', 'email_primary.on_hold')
+          ->addWhere('id', '=', $contactId)
+          ->setLimit(1)
+          ->execute()->first();
+        if (!$contact || $contact['do_not_email'] || empty($contact['email_primary.email']) || \CRM_Utils_Array::value('is_deceased', $contact) || $contact['email_primary.on_hold'] || $contact['is_deleted']) {
+          /*
+           * Contact is deceased or has opted out from mailings so do not send the email
+           */
+          continue;
+        }
+        else {
+          $toName = $contact['display_name'];
+          $toEmail = $contact['email_primary.email'];
+        }
+
+        $schema['contactId'] = 'contactId';
+        $context['contactId'] = $contactId;
+
+        // Whether to enable Smarty evaluation.
+        $useSmarty = (defined('CIVICRM_MAIL_SMARTY') && CIVICRM_MAIL_SMARTY) ?? FALSE;
+
+        $tokenProcessor = new TokenProcessor(\Civi::dispatcher(), [
+          'controller' => __CLASS__,
+          'schema' => $schema,
+          'smarty' => $useSmarty,
+        ]);
+
+        // Populate the token processor.
+        $tokenProcessor->addMessage('messageSubject', ($this->subject ?? $messageTemplates->msg_subject), 'text/plain');
+        $tokenProcessor->addMessage('html', $messageTemplates->msg_html, 'text/html');
+        $tokenProcessor->addMessage('text', ($messageTemplates->msg_text ?? \CRM_Utils_String::htmlToText($messageTemplates->msg_html)), 'text/plain');
+        $row = $tokenProcessor->addRow($context);
+        // Evaluate and render.
+        $tokenProcessor->evaluate();
+        $messageSubject = $row->render('messageSubject');
+        $html = $row->render('html');
+        $text = $row->render('text');
+
+        // set up the parameters for CRM_Utils_Mail::send
+        $mailParams = [
+          'groupName' => 'Email from API',
+          'from' => $from,
+          'toName' => $toName,
+          'toEmail' => $toEmail,
+          'subject' => $messageSubject,
+          'messageTemplateID' => $messageTemplates->id,
+          'contactId' => $contactId,
+          'attachments' => [$attachment],
+        ];
+
+        // render the &amp; entities in text mode, so that the links work
+        $mailParams['text'] = str_replace('&amp;', '&', $text);
```

Comment:
This seems incorrect. It seems to be that it's `CRM_Utils_String::htmlToText` job to decode HTML entities. If decoding `&amp;` seems needed here, it may be an indication that encoding is not correctly handled somewhere else.